src/config_manager.cpp

```diff
@@ -0,0 +1,79 @@
+#include "config_manager.h"
+#include "rosflight.h"
+
+namespace rosflight_firmware
+{
+ConfigManager::ConfigManager(ROSflight &RF, config_t &config):
+  RF_{RF},
+  config_{config}
+{}
+
+bool ConfigManager::init()
+{
+  if(!read())
+    fill_defaults();
+  return true;
+}
+
+bool ConfigManager::configure_devices()
+{
+  bool success = true;
+  for(device_t device{static_cast<device_t>(0)}; device < Configuration::DEVICE_COUNT; ++device)
+    success = success &&RF_.board_.enable_device(device, config_.config[device], RF_.params_);
+  return success;
```

Comment:
It looks like we have an opportunity here to catch configuration errors and not allow the flight controller to arm if they exist. This could probably happen in a future pull request, as it would take some thinking and refactoring. But we could have the configuration manager report error and error resolutions to the state manager to control arming, like what we do with RC, etc.